File: MiniDOMFinanziaria/Finanziaria/Tabelle Finanziarie/CTabelleFinanziarieCursor.cs
using System;
using DMD;
using DMD.XML;
using minidom;
using static minidom.Sistema;


namespace minidom
{
    public partial class Finanziaria
    {

        /// <summary>
        /// Cursore per muoversi all'interno della tabella delle tabelle finanziarie
        /// </summary>
        /// <remarks></remarks>
        [Serializable]
        public class CTabelleFinanziarieCursor 
            : Databases.DBObjectCursor<CTabellaFinanziaria>
        {
            private DBCursorField<int> m_CessionarioID = new DBCursorField<int>("Cessionario");
            private DBCursorStringField m_NomeCessionario = new DBCursorStringField("NomeCessionario");
            private DBCursorStringField m_Nome = new DBCursorStringField("Nome");
            private DBCursorStringField m_Descrizione = new DBCursorStringField("Descrizione");
            private DBCursorField<bool> m_TANVariabile = new DBCursorField<bool>("TANVariabile");
            private DBCursorField<double> m_TAN = new DBCursorField<double>("TAN");
            private DBCursorField<DateTime> m_DataInizio = new DBCursorField<DateTime>("DataInizio");
            private DBCursorField<DateTime> m_DataFine = new DBCursorField<DateTime>("DataFine");
            private DBCursorField<bool> m_Visible = new DBCursorField<bool>("Visible");
            private DBCursorField<double> m_ProvvMax = new DBCursorField<double>("ProvvMax");
            private DBCursorField<double> m_ProvvMaxConRinnovi = new DBCursorField<double>("ProvvMaxRinn");
            private DBCursorField<double> m_ProvvMaxConEstinzioni = new DBCursorField<double>("ProvvMaxEst");
            private DBCursorField<double> m_Sconto = new DBCursorField<double>("Sconto");
            private DBCursorField<TabellaFinanziariaFlags> m_Flags = new DBCursorField<TabellaFinanziariaFlags>("Flags");
            private DBCursorField<decimal> m_UpFrontMax = new DBCursorField<decimal>("UpFrontMax");
            private DBCursorField<TipoCalcoloProvvigioni> m_TipoCalcoloProvvigioni = new DBCursorField<TipoCalcoloProvvigioni>("TipoCalcoloProvvigioni");
            private DBCursorStringField m_FormulaProvvigioni = new DBCursorStringField("FormulaProvvigioni");
            private DBCursorField<double> m_ScontoVisibile = new DBCursorField<double>("ScontoVisibile");
            private DBCursorField<double> m_ProvvAggVisib = new DBCursorField<double>("ProvvAggVisib");
            private DBCursorField<TipoCalcoloProvvigioni> m_TipoCalcoloProvvTAN = new DBCursorField<TipoCalcoloProvvigioni>("TipoCalcoloProvvTAN");
            private DBCursorField<double> m_ProvvTANR = new DBCursorField<double>("ProvvTANR");
            private DBCursorField<double> m_ProvvTANE = new DBCursorField<double>("ProvvTANE");
            private DBCursorStringField m_Categoria = new DBCursorStringField("Categoria");
            private bool m_OnlyValid;


            /// <summary>
            /// Costruttore
            /// </summary>
            public CTabelleFinanziarieCursor()
            {
                m_OnlyValid = false;
            }

            public DBCursorStringField Categoria
            {
                get
                {
                    return m_Categoria;
                }
            }

            /// <summary>
        /// Provvigione TAN in caso di rinnovi
        /// </summary>
        /// <returns></returns>
            public DBCursorField<double> ProvvTANR
            {
                get
                {
                    return m_ProvvTANR;
                }
            }

            /// <summary>
        /// Provvigione TAN in caso di estinzioni
        /// </summary>
        /// <returns></returns>
            public DBCursorField<double> ProvvTANE
            {
                get
                {
                    return m_ProvvTANE;
                }
            }

            public DBCursorField<TipoCalcoloProvvigioni> TipoCalcoloProvvTAN
            {
                get
                {
                    return m_TipoCalcoloProvvTAN;
                }
            }

            /// <summary>
        /// Provvigione aggiuntiva visibile
        /// </summary>
        /// <returns></returns>
            public DBCursorField<double> ProvvAggVisib
            {
                get
                {
                    return m_ProvvAggVisib;
                }
            }

            protected override Sistema.CModule GetModule()
            {
                return TabelleFinanziarie.Module; // ("modCQSPDTblFinanz")
            }

            public bool OnlyValid
            {
                get
                {
                    return m_OnlyValid;
                }

                set
                {
                    m_OnlyValid = value;
                }
            }

            public DBCursorField<TipoCalcoloProvvigioni> TipoCalcoloProvvigioni
            {
                get
                {
                    return m_TipoCalcoloProvvigioni;
                }
            }

            public DBCursorStringField FormulaProvvigioni
            {
                get
                {
                    return m_FormulaProvvigioni;
                }
            }

            public DBCursorField<double> Sconto
            {
                get
                {
                    return m_Sconto;
                }
            }

            public DBCursorField<TabellaFinanziariaFlags> Flags
            {
                get
                {
                    return m_Flags;
                }
            }

            public DBCursorField<double> ProvvMax
            {
                get
                {
                    return m_ProvvMax;
                }
            }

            public DBCursorField<bool> Visible
            {
                get
                {
                    return m_Visible;
                }
            }

            public DBCursorField<int> CessionarioID
            {
                get
                {
                    return m_CessionarioID;
                }
            }

            public DBCursorStringField NomeCessionario
            {
                get
                {
                    return m_NomeCessionario;
                }
            }

            public DBCursorStringField Nome
            {
                get
                {
                    return m_Nome;
                }
            }

            public DBCursorStringField Descrizione
            {
                get
                {
                    return m_Descrizione;
                }
            }

            public DBCursorField<bool> TANVariabile
            {
                get
                {
                    return m_TANVariabile;
                }
            }

            public DBCursorField<double> TAN
            {
                get
                {
                    return m_TAN;
                }
            }

            public DBCursorField<DateTime> DataInizio
            {
                get
                {
                    return m_DataInizio;
                }
            }

            public DBCursorField<DateTime> DataFine
            {
                get
                {
                    return m_DataFine;
                }
            }

            public DBCursorField<double> ProvvMaxConRinnovi
            {
                get
                {
                    return m_ProvvMaxConRinnovi;
                }
            }

            public DBCursorField<double> ProvvMaxConEstinzioni
            {
                get
                {
                    return m_ProvvMaxConEstinzioni;
                }
            }

            public override string GetTableName()
            {
                return "tbl_FIN_TblFin";
            }

            public override object InstantiateNew(DBReader dbRis)
            {
                return new CTabellaFinanziaria();
            }

            protected override Databases.CDBConnection GetConnection()
            {
                return Database;
            }

            protected override void XMLSerialize(XMLWriter writer)
            {
                writer.WriteAttribute("OnlyValid", m_OnlyValid);
                base.XMLSerialize(writer);
            }

            protected override void SetFieldInternal(string fieldName, object fieldValue)
            {
                switch (fieldName ?? "")
                {
                    case "OnlyValid":
                        {
                            m_OnlyValid = DMD.XML.Utils.Serializer.DeserializeBoolean(fieldValue) == true;
                            break;
                        }

                    default:
                        {
                            base.SetFieldInternal(fieldName, fieldValue);
                            break;
                        }
                }
            }

            public DBCursorField<double> ScontoVisibile
            {
                get
                {
                    return m_ScontoVisibile;
                }
            }

            public override string GetWherePart()
            {
                string wherePart = base.GetWherePart();
                if (m_OnlyValid)
                {
                    wherePart = DMD.Strings.Combine(wherePart, "(([DataInizio] Is Null) Or ([DataInizio]<=" + DBUtils.DBDate(DMD.DateUtils.ToDay()) + "))", " AND ");
                    wherePart = DMD.Strings.Combine(wherePart, "(([DataFine] Is Null) Or ([DataFine]>=" + DBUtils.DBDate(DMD.DateUtils.ToDay()) + "))", " AND ");
                }

                return wherePart;
            }
        }
    }
}

File: MiniDOMFinanziaria/Finanziaria/Tabelle Finanziarie/CTabelleFinanziarieCursor.vb
Imports minidom.Databases
Imports minidom.Sistema

Partial Public Class Finanziaria

    ''' <summary>
    ''' Cursore per muoversi all'interno della tabella delle tabelle finanziarie
    ''' </summary>
    ''' <remarks></remarks>
    Public Class CTabelleFinanziarieCursor
        Inherits DBObjectCursor(Of CTabellaFinanziaria)

        Private m_CessionarioID As New CCursorField(Of Integer)("Cessionario")
        Private m_NomeCessionario As New CCursorFieldObj(Of String)("NomeCessionario")
        Private m_Nome As New CCursorFieldObj(Of String)("Nome")
        Private m_Descrizione As New CCursorFieldObj(Of String)("Descrizione")
        Private m_TANVariabile As New CCursorField(Of Boolean)("TANVariabile")
        Private m_TAN As New CCursorField(Of Double)("TAN")
        Private m_DataInizio As New CCursorField(Of Date)("DataInizio")
        Private m_DataFine As New CCursorField(Of Date)("DataFine")
        Private m_Visible As New CCursorField(Of Boolean)("Visible")
        Private m_ProvvMax As New CCursorField(Of Double)("ProvvMax")
        Private m_ProvvMaxConRinnovi As New CCursorField(Of Double)("ProvvMaxRinn")
        Private m_ProvvMaxConEstinzioni As New CCursorField(Of Double)("ProvvMaxEst")
        Private m_Sconto As New CCursorField(Of Double)("Sconto")
        Private m_Flags As New CCursorField(Of TabellaFinanziariaFlags)("Flags")
        Private m_UpFrontMax As New CCursorField(Of Decimal)("UpFrontMax")
        Private m_TipoCalcoloProvvigioni As New CCursorField(Of TipoCalcoloProvvigioni)("TipoCalcoloProvvigioni")
        Private m_FormulaProvvigioni As New CCursorFieldObj(Of String)("FormulaProvvigioni")
        Private m_ScontoVisibile As New CCursorField(Of Double)("ScontoVisibile")
        Private m_ProvvAggVisib As New CCursorField(Of Double)("ProvvAggVisib")
        Private m_TipoCalcoloProvvTAN As New CCursorField(Of TipoCalcoloProvvigioni)("TipoCalcoloProvvTAN")
        Private m_ProvvTANR As New CCursorField(Of Double)("ProvvTANR")
        Private m_ProvvTANE As New CCursorField(Of Double)("ProvvTANE")
        Private m_Categoria As New CCursorFieldObj(Of String)("Categoria")
        Private m_OnlyValid As Boolean

        Public Sub New()
            Me.m_OnlyValid = False
        End Sub

        Public ReadOnly Property Categoria As CCursorFieldObj(Of String)
            Get
                Return Me.m_Categoria
            End Get
        End Property

        ''' <summary>
        ''' Provvigione TAN in caso di rinnovi
        ''' </summary>
        ''' <returns></returns>
        Public ReadOnly Property ProvvTANR As CCursorField(Of Double)
            Get
                Return Me.m_ProvvTANR
            End Get
        End Property

        ''' <summary>
        ''' Provvigione TAN in caso di estinzioni
        ''' </summary>
        ''' <returns></returns>
        Public ReadOnly Property ProvvTANE As CCursorField(Of Double)
            Get
                Return Me.m_ProvvTANE
            End Get
        End Property

        Public ReadOnly Property TipoCalcoloProvvTAN As CCursorField(Of TipoCalcoloProvvigioni)
            Get
                Return Me.m_TipoCalcoloProvvTAN
            End Get
        End Property

        ''' <summary>
        ''' Provvigione aggiuntiva visibile
        ''' </summary>
        ''' <returns></returns>
        Public ReadOnly Property ProvvAggVisib As CCursorField(Of Double)
            Get
                Return Me.m_ProvvAggVisib
            End Get
        End Property

        Protected Overrides Function GetModule() As CModule
            Return minidom.Finanziaria.TabelleFinanziarie.Module '("modCQSPDTblFinanz")
        End Function

        Public Property OnlyValid As Boolean
            Get
                Return Me.m_OnlyValid
            End Get
            Set(value As Boolean)
                Me.m_OnlyValid = value
            End Set
        End Property

        Public ReadOnly Property TipoCalcoloProvvigioni As CCursorField(Of TipoCalcoloProvvigioni)
            Get
                Return Me.m_TipoCalcoloProvvigioni
            End Get
        End Property

        Public ReadOnly Property FormulaProvvigioni As CCursorFieldObj(Of String)
            Get
                Return Me.m_FormulaProvvigioni
            End Get
        End Property

        Public ReadOnly Property Sconto As CCursorField(Of Double)
            Get
                Return Me.m_Sconto
            End Get
        End Property

        Public ReadOnly Property Flags As CCursorField(Of TabellaFinanziariaFlags)
            Get
                Return Me.m_Flags
            End Get
        End Property

        Public ReadOnly Property ProvvMax As CCursorField(Of Double)
            Get
                Return Me.m_ProvvMax
            End Get
        End Property

        Public ReadOnly Property Visible As CCursorField(Of Boolean)
            Get
                Return Me.m_Visible
            End Get
        End Property

        Public ReadOnly Property CessionarioID As CCursorField(Of Integer)
            Get
                Return Me.m_CessionarioID
            End Get
        End Property

        Public ReadOnly Property NomeCessionario As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeCessionario
            End Get
        End Property

        Public ReadOnly Property Nome As CCursorFieldObj(Of String)
            Get
                Return Me.m_Nome
            End Get
        End Property

        Public ReadOnly Property Descrizione As CCursorFieldObj(Of String)
            Get
                Return Me.m_Descrizione
            End Get
        End Property

        Public ReadOnly Property TANVariabile As CCursorField(Of Boolean)
            Get
                Return Me.m_TANVariabile
            End Get
        End Property

        Public ReadOnly Property TAN As CCursorField(Of Double)
            Get
                Return Me.m_TAN
            End Get
        End Property

        Public ReadOnly Property DataInizio As CCursorField(Of Date)
            Get
                Return Me.m_DataInizio
            End Get
        End Property

        Public ReadOnly Property DataFine As CCursorField(Of Date)
            Get
                Return Me.m_DataFine
            End Get
        End Property

        Public ReadOnly Property ProvvMaxConRinnovi As CCursorField(Of Double)
            Get
                Return Me.m_ProvvMaxConRinnovi
            End Get
        End Property

        Public ReadOnly Property ProvvMaxConEstinzioni As CCursorField(Of Double)
            Get
                Return Me.m_ProvvMaxConEstinzioni
            End Get
        End Property

        Public Overrides Function GetTableName() As String
            Return "tbl_FIN_TblFin"
        End Function

        Public Overrides Function InstantiateNew(dbRis As DBReader) As Object
            Return New CTabellaFinanziaria
        End Function

        Protected Overrides Function GetConnection() As CDBConnection
            Return Finanziaria.Database
        End Function

        Protected Overrides Sub XMLSerialize(writer As XML.XMLWriter)
            writer.WriteAttribute("OnlyValid", Me.m_OnlyValid)
            MyBase.XMLSerialize(writer)
        End Sub

        Protected Overrides Sub SetFieldInternal(fieldName As String, fieldValue As Object)
            Select Case fieldName
                Case "OnlyValid" : Me.m_OnlyValid = XML.Utils.Serializer.DeserializeBoolean(fieldValue)
                Case Else : MyBase.SetFieldInternal(fieldName, fieldValue)
            End Select
        End Sub

        Public ReadOnly Property ScontoVisibile As CCursorField(Of Double)
            Get
                Return Me.m_ScontoVisibile
            End Get
        End Property

        Public Overrides Function GetWherePart() As String
            Dim wherePart As String = MyBase.GetWherePart()
            If Me.m_OnlyValid Then
                wherePart = Strings.Combine(wherePart, "(([DataInizio] Is Null) Or ([DataInizio]<=" & DBUtils.DBDate(DateUtils.ToDay) & "))", " AND ")
                wherePart = Strings.Combine(wherePart, "(([DataFine] Is Null) Or ([DataFine]>=" & DBUtils.DBDate(DateUtils.ToDay) & "))", " AND ")
            End If
            Return wherePart
        End Function
    End Class

End Class